test/testenv/secretutil.go

```diff
@@ -88,53 +88,63 @@ func GetSecretKey(deployment *Deployment, ns string, key string, secretName stri
 	}
 	logf.Log.Info("Get secret object encoded value", "Secret Name", secretName, "Key", key)
 	value := "Invalid Key"
-	if key == "hec_token" {
+	switch key {
+	case "hec_token":
 		value = DecodeBase64(restResponse.Data.HecToken)
-	}
-	if key == "idxc_secret" {
+	case "idxc_secret":
 		value = DecodeBase64(restResponse.Data.IdxcSecret)
-	}
-	if key == "pass4SymmKey" {
+	case "pass4SymmKey":
 		value = DecodeBase64(restResponse.Data.Pass4SymmKey)
-	}
-	if key == "password" {
+	case "password":
 		value = DecodeBase64(restResponse.Data.Password)
-	}
-	if key == "shc_secret" {
+	case "shc_secret":
 		value = DecodeBase64(restResponse.Data.ShcSecret)
 	}
 	return value
 }
 
 //ModifySecretObject Modifies the entire secret object
-func ModifySecretObject(deployment *Deployment, data map[string][]byte, ns string, secretName string) bool {
+func ModifySecretObject(deployment *Deployment, data map[string][]byte, ns string, secretName string) error {
 	logf.Log.Info("Modify secret object", "Secret Name", secretName, "Data", data)
 	secret := newSecretSpec(ns, secretName, data)
 	//Update object using spec
 	err := deployment.UpdateCR(secret)
 	if err != nil {
 		logf.Log.Error(err, "Unable to update secret object")
-		return false
+		return err
+	}
+	return nil
+}
+
+//DeleteSecretObject Modifies the entire secret object
+func DeleteSecretObject(deployment *Deployment, data map[string][]byte, ns string, secretName string) error {
+	logf.Log.Info("Delete secret object", "Secret Name", secretName)
+	secret := newSecretSpec(ns, secretName, data)
+	//Update object using spec
+	err := deployment.DeleteCR(secret)
```

Comment:
Do we need the data map here to delete the secret object?. If not it might be beneficial to remove the map from parameter list and just pass empty map like `secret := newSecretSpec(ns, secretName, map[string][]byte)`

Reply:
Removed data as a parameter. Creating data as empty object within the function